ldms/src/store/hello_stream/l2_stream_csv_store.c

```diff
@@ -0,0 +1,1165 @@
+                pthread_mutex_unlock(&cfg_lock);
+                return -1;
+        }
+
+	s = av_value(avl, "buffer");
+	if (!s){
+		buffer = atoi(s);
+		msglog(LDMSD_LDEBUG,
+                       PNAME ": setting buffer to '%d'\n", buffer);
+	}
+
+	s = av_value(avl, "stream");
+	if (!s){
+		msglog(LDMSD_LDEBUG, PNAME ": missing stream in config\n");
+		rc = EINVAL;
+		goto out;
+	} else {
+		streamlist = strdup(s);
+	}
+
+	s = av_value(avl, "path");
+	if (!s){
+		msglog(LDMSD_LDEBUG, PNAME ": missing path in config\n");
+		rc = EINVAL;
+		goto out;
+	} else {
+		root_path = strdup(s);
+		msglog(LDMSD_LDEBUG,
+                       PNAME ": setting root_path to '%s'\n", root_path);
+	}
+
+
+	s = av_value(avl, "container");
+	if (!s){
+		msglog(LDMSD_LDEBUG, PNAME ": missing container in config\n");
+		rc = EINVAL;
+		goto out;
+	} else {
+		container = strdup(s);
+		msglog(LDMSD_LDEBUG,
+                       PNAME ": setting container to '%s'\n", container);
+	}
+
+
+        templist = strdup(streamlist);
+        if (!templist){
+                rc = ENOMEM;
+                goto out;
+        }
+        pch = strtok_r(templist, ",", &saveptr);
+        while (pch != NULL){
+                msglog(LDMSD_LDEBUG,
+                       PNAME ": opening streamstore for '%s'\n", pch);
+                rc = open_streamstore(pch);
+                if (rc){
+                        msglog(LDMSD_LERROR,
+                               PNAME ": Error opening store for strean '%s'\n",
+                               pch);
+                        goto err;
+                }
+                pch = strtok_r(NULL, ",", &saveptr);
+	}
+        free(templist);
+        templist = NULL;
+
+        //only set rollover once we have created the files
+        s = av_value(avl, "rollover");
+        if (s) {
+                rollover = atoi(s);
+                rolltype = ROLL_BY_RECORDS;
+                pthread_create(&rothread, NULL, rolloverThreadInit, NULL);
+                rothread_used = 1;
+                msglog(LDMSD_LDEBUG,
+                       PNAME ": setting rollrecords to %d\n", rollover);
+        }
+
+
+        //subscribe to each one
+        templist = strdup(streamlist);
+        if (!templist){
+                rc = ENOMEM;
+                goto out;
+        }
+        pch = strtok_r(templist, ",", &saveptr);
+        while (pch != NULL){
+                msglog(LDMSD_LDEBUG,
+                       PNAME ": subscribing to stream '%s'\n", pch);
+                ldmsd_stream_subscribe(pch, stream_cb, self);
+                pch = strtok_r(NULL, ",", &saveptr);
+	}
+        free(templist);
+        templist = NULL;
+
+        goto out;
+
+err:
+        //delete any created streamstores
+        idx_traverse(stream_idx, close_streamstore, NULL);
+        idx_destroy(stream_idx);
+        stream_idx = idx_create();
+
+out:
+        if (templist) free(templist);
```

Comment:
remove if (templist) and if (streamlist) tests before free. free accepts NULL correctly.